Topics/Tech_Stacks/Vue_Component_Setup_and_Tips

```diff
@@ -0,0 +1,73 @@
+# Setting up Vue 3 Components, How to Use them, and Other Tips!
+
+## Table of Contents
+
+#### [Introduction](#Introduction)
+#### [What is a Vue Component? ](#what)
+#### [Topics](#thing)
+#### [Topics again](#something)
+#### [Other Tips](#other)
+#### [Conclusion](#conclude)
+
+## Introduction
+This is a guide to the inner workings of Vue Components and their usage. It will tell you how to set them up,
+what the file format is, and also give general advice and things to watch out for. This guide assumes the reader has basic knowledge of HTML/CSS, as well as Javascript.
+
+If you have no knowledge of Vue or how to set up a Vue project, it is imperative that you follow [Vue's Quick Start](https://vuejs.org/guide/quick-start.html) instructions for doing so.
+
+## What is a Vue Component? Why should I care? <a name="what"></a>
+The Vue.js framework is a Javascript component-based programming model for the development of User Interfaces (UI), and is commonly used for Web Applications. It builds on the syntax of HTML, allowing the user to "declaratively" describe what they want the code to do. This allows for a very reactive app, one that responds efficiently to state changes to update the DOM. What this means is, less work needs to be done by the programmer for the HTML to update. The API takes care of a lot of it!
+
+Components are how it's all done. Every .vue file is a Vue Component! In these files, you can condense the 3 key aspects of a webpage, the HTML, code-behind, and CSS styling. This centralization and user-friendliness has made Vue renowned for its simplicity.
+
+**Am I quoting too much from official documentation? Should I put my own spin on it or cut out general info about the Vue.js framework?**
```

Comment:
Honestly, it's fine as-is I think.
On putting your own spin on it -- this isn't an essay, we're trying to teach people how to use Vue, and the official docs are really good at that so emulating them makes sense.

If you want to keep your guide laser focused (and you probably should) -- removing the general info would be sensible. We assume after all the user has a broad sense of what Vue is and how to set it up.